Transform_ Transformer 모듈 생성

bbang7 opened the pull request:
### 세부사항
1. 선택된 물체의 기즈모 변경을 하는 기능을 가진 모듈 및 클래스 생성
  -  축을 이용한 물체의 위치이동 함수
  -  마우스의 위치로의 물체 위치이동 함수
  -  물체 회전 함수
  -  물체의 스케일 변경 함수
  -  EditorGrid 의 그리드 스냅 동기화를 확인하는 함수

bbang7 committed:
Transform_ Transformer 모듈 생성

# 세부사항
1. 선택된 물체의 기즈모 변경을 하는 기능을 가진 모듈 및 클래스 생성
  -  축을 이용한 물체의 위치이동 함수
  -  마우스의 위치로의 물체 위치이동 함수
  -  물체 회전 함수
  -  물체의 스케일 변경 함수
  -  EditorGrid 의 그리드 스냅 동기화를 확인하는 함수

Minusi commented on UEPrototype/Source/UEPrototype/Private/Transformer/Transformer.cpp:
```diff
@@ -0,0 +1,132 @@
+
+// Called every frame
+void ATransformer::Tick(float DeltaTime)
+{
```
사용되지 않는 Tick 함수 삭제 요망

Minusi commented on UEPrototype/Source/UEPrototype/Private/Transformer/Transformer.cpp:
```diff
@@ -0,0 +1,132 @@
+
+// Called when the game starts or when spawned
+void ATransformer::BeginPlay()
+{
```
사용되지 않는 BeginPlay 함수 삭제 요망

Minusi commented on UEPrototype/Source/UEPrototype/Private/Transformer/Transformer.cpp:
```diff
@@ -0,0 +1,132 @@
+void ATransformer::MoveObject(AActor* Target, FVector Direction, float SnapInterval, float Delta)
+{
+	FSnapStruct SnapElement;
+
```
포인터의 null 유효성 검증할 것
ex)
if(IsValid(Tartget) == false) return;

과 같이 유효성 검증 반드시 수행할 것

Minusi commented on UEPrototype/Source/UEPrototype/Private/Transformer/Transformer.cpp:
```diff
@@ -0,0 +1,132 @@
+
+	Snap(Delta, SnapInterval, SnapElement);
+
+	if (SnapElement.bCanSnap) Target->SetActorLocation(Target->GetActorLocation() + Direction * SnapElement.SnappedDelta);
```
두 줄로 나눌 것

Minusi commented on UEPrototype/Source/UEPrototype/Private/Transformer/Transformer.cpp:
```diff
@@ -0,0 +1,132 @@
+
+	Snap(Delta, SnapInterval, SnapElement);
+
+	if (SnapElement.bCanSnap) Target->AddActorWorldRotation(UKismetMathLibrary::RotatorFromAxisAndAngle(Axis, SnapElement.SnappedDelta * -1));
```
두 줄로 나눌 것

Minusi commented on UEPrototype/Source/UEPrototype/Public/Transformer/Transformer.h:
```diff
@@ -0,0 +1,87 @@
+
+};
+
+UCLASS()
```
클래스의 용도와 목적에 대한 주석 작성할 것

Minusi commented on UEPrototype/Source/UEPrototype/Private/Transformer/Transformer.cpp:
```diff
@@ -0,0 +1,132 @@
+
+void ATransformer::SyncWithGridSnap(bool bIsSyncWithGridSnap)
+{
+	this->bIsSyncWithGridSnap = bIsSyncWithGridSnap;
```
재량껏 수정할 것

Minusi commented on UEPrototype/Source/UEPrototype/Public/Transformer/Transformer.h:
```diff
@@ -0,0 +1,87 @@
+
+
+protected:
+	// Called when the game starts or when spawned
```
마찬가지로 BeginPlay 함수 삭제할 것

Minusi commented on UEPrototype/Source/UEPrototype/Public/Transformer/Transformer.h:
```diff
@@ -0,0 +1,87 @@
+
+public:
+	// Called every frame
+	virtual void Tick(float DeltaTime) override;
```
마찬가지로 Tick 함수 삭제할 것

bbang7 reviewed (commented):
코멘트 수정 완료

bbang7 committed:
Transform_ 풀 리퀘스트 코멘트 기반 코드 수정
# 세부사항
1. 함수의 유효성 검사
  -  MoveObject, MoveObjectToCurser, RotateObject, ScaleObject 함수의 유효성 검사 완료
2. SyncWithGridSnap 매개변수 이름 변경
3. Transformer.h 클래스 파일 주석 설명 기재.
4. 불필요한 함수 (BeginPlay, Tick) 제거
